zone/aa.cpp

```diff
@@ -918,6 +926,54 @@ void Client::SendAlternateAdvancementRank(int aa_id, int level) {
 	safe_delete(outapp);
 }
 
+void Client::SendPhantomStatsAlternateAdvancementRank() {
+	//We only need to send phantom stats if stats are set to a custom value
+	if (RuleI(Character, StatCap) <= 0) {
+		return;
+	}
+
+	auto diff = RuleI(Character, StatCap) - GetMaxStat(false);
+	int size = sizeof(AARankInfo_Struct) + (sizeof(AARankEffect_Struct) * 7);
+	auto outapp = new EQApplicationPacket(OP_SendAATable, size);
+	AARankInfo_Struct *aai = (AARankInfo_Struct*)outapp->pBuffer;
+
+	aai->id = detail::PhantomStatId;
+	aai->upper_hotkey_sid = -1;
+	aai->lower_hotkey_sid = -1;
+	aai->title_sid = detail::PhantomStatId;
+	aai->desc_sid = detail::PhantomStatId;
+	aai->cost = 1;
+	aai->seq = detail::PhantomStatId;
+	aai->type = 3;
+	aai->spell = -1;
+	aai->spell_type = 0;
+	aai->spell_refresh = 0;
+	aai->classes = 16777215;
+	aai->level_req = 1;
+	aai->current_level = 2;
+	aai->max_level = 1;
+	aai->prev_id = -1;
+	aai->next_id = -1;
+	aai->total_cost = 1;
+	aai->expansion = 0;
+	aai->category = -1;
+	aai->charges = 0;
+	aai->grant_only = 1;
+	aai->total_effects = 7;
+	aai->total_prereqs = 0;
+
+	outapp->SetWritePosition(sizeof(AARankInfo_Struct));
+	for (int i = 0; i < 7; ++i) {
+		outapp->WriteSInt32(SE_RaiseStatCap);
+		outapp->WriteSInt32(diff);
+		outapp->WriteSInt32(0);
+		outapp->WriteSInt32(i + 1);
```

Comment:
outapp->WriteSInt32(SE_RaiseStatCap);
		outapp->WriteSInt32(diff);
		outapp->WriteSInt32(i + 1);
                outapp->WriteSInt32(0);

these are flipped and this corrects it

Reply:
Based on existing AA code what I have is correct the order is:
```
outapp->WriteSInt32(effect.effect_id);
outapp->WriteSInt32(effect.base1);
outapp->WriteSInt32(effect.base2);
outapp->WriteSInt32(effect.slot);
```

Reply:
Yes and base2 is the "stattype" for that AA..  so the base2 always being 0 will give all the stat raises to "strength" only screenshot below
![Screenshot_64](https://user-images.githubusercontent.com/3869308/134212049-340d0b91-e6d7-4aaf-89f6-e87b8c6d040a.png)



![Screenshot_72](https://user-images.githubusercontent.com/3869308/134212507-b2a2458f-ea38-4d5f-964a-71b1e4b9eb14.png)


Check out the Planar power AA on PeqEditor base2 is the stattype for StatCap

Reply:
Now that I think about it, changing it to just 

(i)  and not +1 would be better fix and leaving it ...

0; i < 7

Because stattype starts from 0 anyways

Reply:
Ah I see what you mean now, it isn't so much swapped as it is missing.